cookbooks/fb_bookworm/files/default/bookworm/crawler.rb

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2022-present, Meta, Inc.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require 'rubocop'
+require 'parser/current'
+# require 'ripper'
+
+# TODO(dcrosby) move AST generation to a rule? This would allow multiple AST
+# parsers (ie ripper), which would be beneficial on heavy-duty rules It would
+# also remove overhead if only using string search/regexes in a rule (not a lot
+# of great use cases, but they exist)
+
+module Bookworm
+  class Crawler
+    attr_reader :processed_files
+
+    def initialize(config, keys: [])
+      @config = config
+
+      # TODO(dcrosby) add messages to verbose mode
+      to_crawl = {}
+      keys.each do |k|
+        v = BOOKWORM_KEYS[k]
+        to_crawl[k] = @config.source_dirs[v['source_dirs']].
+                      map { |d| Dir.glob("#{d}/#{v['glob_pattern']}") }.flatten
+      end
+      @processed_files = {}
+      to_crawl.each do |key, files|
+        instance_variable_set("@#{key}_intake_queue", files)
+        instance_variable_set("@#{key}_processed_files", {})
```

Comment:
why that instead of:
```
@intake_queue[key] = files
```
?
And `@#{key}_processed_files` doesn't seem to ever be used...

Reply:
Used `@intake_queue[key]` suggestion and dropped unused variable in 31aa98e0b4841459f38ecec58a3713d1f2f2928b